arrow/src/array/array_list.rs

```diff
@@ -50,6 +50,9 @@ impl OffsetSizeTrait for i64 {
     }
 }
 
+/// Generic struct for a primitive Array
+///
+/// Instead of using `GenericListArray` directly, consider using `ListArray` or `LargeListArray`
```

Comment:
```suggestion
/// Instead of using `GenericListArray` directly, consider using [`ListArray`] or [`LargeListArray`]
```

If you use ```[`ListArray`]``` instead, the docs will include a hyperlink to the class

Reply:
fwiw, I do not think this recommendation is fair. `GenericListArray` allows to write code generic over the `Offset`, which is more generic than only supporting `ListArray` or `LArgeListArray`.

The same applies for `Utf8Array<T>`, `PrimitiveArray<T>`, etc.

Reply:
Thanks @jorgecarleitao - I'm a little confused though. My initial commit attempted to show an example of this https://github.com/apache/arrow-rs/pull/474/commits/4c5b2b45ceabfdc0564a46eed15bf937333dbb91 which admittedly was the equivalent of the example for `ListArray` 

Would you perhaps suggest different wording for this comment, or perhaps could you point me in the direction of a better example that we could use for `GenericListArray` ?

Reply:
I am sorry I was not very explicit. :(

The example is great, I would just omit the 

> Instead of using `GenericListArray` directly, consider using [`ListArray`] or [`LargeListArray`]`

which suggests that using `ListArray` is better than using `GenericListArray`, when imo we do not have enough information here to make this recommendation. In general using `GenericListArray` is better because it allows to write things like

```
fn some_f<O: ListOffset>(array: &GenericListArray<O>)
```

which makes it generic over both "ListArray" and "LargeListArray". Most code in arrow and DataFusion uses this to reduce code duplication while supporting both.

Reply:
Thanks - fwiw I was trying to tackle the specific problem of showing by an example, how to construct a `GenericListArray` 

Per https://github.com/apache/arrow-rs/pull/474#pullrequestreview-687894203 I removed the example for `GenericListArray` in favour of examples for `ListArray` and `LargeListArray.` The intent was not to suggest that using `ListArray` is better or worse than using `GenericListArray` 

In any case, also removing the comment will result in no documentation or examples for `GenericListArray`

Is that the final state we want? Or should we have some documentation or hints for usage for `GenericListArray`

Hope that all makes sense. Thanks.

Reply:
I think it is valuable if the comment on `GenericListArray` pointed (likely) novice users towards `ListArray` and `LargeListArray` as their existence may not be obvious at this stage

How about making the wording less judgemental such as 

```
For non generic lists, you may wish to consider using [`ListArray`] or [`LargeListArray`]`
```